### ISSUE TITLE
Bug: Graphic EQ Audio Unit Plug-in crashes often

### ISSUE THREAD

**Issue by kevinfales** [bitgapp/eqMac#910]
**Disclaimer:** 
Please go through [existing issues](https://github.com/bitgapp/eqMac/issues?q=is%3Aopen+is%3Aissue+label%3A%22type%3A+bug%22) to avoid creating duplicates. If you found a similar issue please post a comment there. If you are uncertain your issue is related to others then create the issue. Obvious ignorance of this rule might get you blocked from posting new issues. Please tick the box below to show that you have followed the guideline.
- [x] I have checked for a similar issue and sure it hasn't been reported before.

**Describe the bug**
Several times per day, eqMac will crash and then re-launch, showing a dialog explaining that the Graphic EQ Audio Unit Plug-in has crashed. Additionally, when this happens, the Graphic EQ is removed from all presets.

**Steps to Reproduce**
1. In the Extra Effects panel, click "Add Effect" and select "Apple: Graphic EQ"
2. Configure the EQ to preference (doubtful the actual values matter here)
3. In the Extra Effects panel, click the [ + ] icon and enter a name for the preset, then Save.
4. After a seemingly random amount of time (typically between 45-90 mins), eqMac will crash, and relaunch with the following dialog: 
![image](https://github.com/bitgapp/eqMac/assets/1454723/d7efbffc-7dc5-436a-8dbf-269199b27795)

Notes:
* Output device seems to have no impact on this occurring. I've experienced it using bluetooth headphones and built-in speakers.
* (I believe) This only began occurring after the update to 1.8.3

**Expected behaviour**
No crashes occur.

**Setup information:**
 - Audio device used for playback: Sony WH-1000XM5, Macbook Pro speakers
 - Audio transmission interface: Bluetooth, Built-in speakers
 - macOS Version: 14.2.1 (Sonoma)
 - eqMac Version: 1.8.7 (beta, but issue exists on the latest stable release (1.8.5) as well)
 - eqMac UI Version: 5.0.8 (Remote)
 - Driver version: 2.5.3

**Comment by kevinfales** [comment]
Can I help debug this somehow? It's incredibly frustrating and happens several times a day

**Comment by dan-iway** [comment]
Just wanted to add that I have the same problem (although I am using the "Apple: Dynamics Processor" Plugin instead).
I understand the crashes may not be easy to fix, an easier fix could possibly be to at least not remove it and keep the presets saved and just disabled it instead.
If i have restart and enable it again it's annoying but much less frustrating than also configuring it again.

**Comment by kevinfales** [comment]
> ...an easier fix in the meantime could possibly be to at least not remove it and keep the presets saved and just disabled it instead.

Agreed that deleting the preset makes the problem orders of magnitude more frustrating, since you have to go build the settings for it all over.

Here's a workaround I've been using to at least sidestep this part of the issue:
1. After adding a plugin (Graphic EQ or Dynamics Processor) in the Extra Effects panel, save the preset as you normally would. 
2. Then, click save again and give it another name (mine are just labelled `v1` and `v1.1`). 
3. When the plugin inevitably crashes, you can switch the dropdown to your second saved preset, where your preset will still exist intact
4. After switching, click the save button again and overwrite the version that just crashed.

This absolutely doesn't solve the entire issue and requires a couple steps whenever this thing crashes, but IMO it's far less work than re-configuring a plugin every time.

**Comment by dan-iway** [comment]
@kevinfales Good tip, thank you, I will use that. Agree fully with what you said about this not solving the issue of course 👍

**Comment by coroiu** [comment]
Seeing the same issue with the CanOpener Studio plugin. Logs have been emailed to customer support.

**Comment by kevinfales** [comment]
**UPDATE**

Updating to MacOS 14.4.1 (latest as of today) seems to address this issue. No longer seeing audio unit crashes.

From the [14.4.1 patch notes](https://www.macrumors.com/2024/03/26/update-your-mac-macos-14-4-1-included-bug-fixes/):
> Audio Unit Plugins Crash
> FIXED
> There have been reports of compatibility issues with PACE products in macOS 14.4, including the [iLok License Manager](https://help.ilok.com/faq_licenses.html#sonoma_14_4_plugins), connected to Apple silicon Macs running macOS 14.4. The problem specifically impacts plugins from various manufacturers that are safeguarded by PACE/iLok, including those [made by Universal Audio](https://help.uaudio.com/hc/en-us/articles/24492336117652-PACE-iLok-Issue-with-macOS-Sonoma-14-4-Update), when used with host DAWs such as Logic Pro. Apple's release notes for macOS 14.4.1 say the update provides a fix for cases in which "copy protected Audio Unit plug-ins designed for professional music apps may not open or pass validation."

**Comment by dan-iway** [comment]
@kevinfales  I am running 14.4.1 as well and just had a crash yesterday :-(

**Comment by coroiu** [comment]
Can confirm, using 14.4.1 and still crashing

**Comment by ragauskl** [comment]
@dan-iway @coroiu Can you check if these plugins crash with any other software besides eqMac? I consulted with the dev and plugin crashes should not be related to/caused by eqMac. Just like with kevinfales case it may be an issue with macOS version compatibility with those plugins or an issue with the plugin itself.

**Comment by dan-iway** [comment]
@ragauskl How are we supposed to test this? The plugin does not crash for hours in eqMac either...
And while it may be true that the plugin crash has nothing to do with eqMac, eqMac could definitely handle this better - just restart the plugin and keep the config.

**Comment by kevinfales** [comment]
@ragauskl I'm on a completely different computer now (Macbook Pro M3) running the latest MacOS (14.5) and I just had a crash of the audio plugin after about an hour.

> eqMac could definitely handle this better - just restart the plugin and keep the config.

Yes. Yes yes. The crash is frustrating, but the fact that it deletes my preset is far, far more frustrating.

**Comment by ragauskl** [comment]
> just restart the plugin

@dan-iway this is how eqMac used to work before but some plugins would continuously crash causing even bigger problem.

Dev will be looking into a better handling of plugin crashes so it's not as annoying, but said that it may not be as easy as simply disabling the crashing plugin if possible at all (don't ask me, I'm just a middleman between you and him, he knows the details but I can't bother him about every single one).

Regarding testing, I know it may not be easy to replicate as it takes hours, but if anyone can use another software and use the plugins to see if they crash in there as well can help with identifying if it's eqMac issue which would help prfioritize this (specifically the crash investigation, better handling of errors like this will be looked into anyway).

**Comment by kevinfales** [comment]
Is there any update or priority on a fix here? This happens probably 15 or 20 times per day and it's a serious vibe-killer.

**Comment by ragauskl** [comment]
For now there's no update on the issue as there are higher priority (issues affecting more users) tasks being worked on, until those are resolved there's no timeline for when this will be worked on. It's in the to-do list, just not at the top of the list.

**Comment by coroiu** [comment]
App v1.8.8
Driver v.2.5.4 
UI: 5.1.4 (Local)
macOS: 15.0

No longer seeing this issue, have been using for two days straight 👍 I also switched from an M1 Pro to M3 Ultra so I'm not sure if that might've been the reason. Either way now I can fully enjoy this awesome app!

**Comment by intence-ch** [comment]
Same here. Upgraded to M4 Pro with MacOS 15 and no crash in about a month

**Comment by dan-iway** [comment]
Same here. Upgraded to M4 Pro with MacOS 15 and no crash in about a month
